server/src/main/java/org/apache/uniffle/server/DirectMemoryUsageReporter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.server;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import io.grpc.netty.shaded.io.netty.util.internal.PlatformDependent;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.uniffle.common.util.ThreadUtils;
+
+public class DirectMemoryUsageReporter {
+
+  private static final Logger LOG = LoggerFactory.getLogger(DirectMemoryUsageReporter.class);
+
+  private final long reportInitialDelay;
+  private final long reportInterval;
+  private final ScheduledExecutorService service =
+      Executors.newSingleThreadScheduledExecutor(
+          ThreadUtils.getThreadFactory("DirectMemoryReporter-%d"));
+
+  public DirectMemoryUsageReporter(ShuffleServerConf conf) {
+    this.reportInitialDelay =
+        conf.getLong(ShuffleServerConf.SERVER_DIRECT_MEMORY_USAGE_REPORT_DELAY);
+    this.reportInterval =
+        conf.getLong(ShuffleServerConf.SERVER_DIRECT_MEMORY_USAGE_REPORT_INTERVAL);
+  }
+
+  public void start() {
+    LOG.info(
+        "Start report direct memory usage to MetricSystem after {}ms and interval is {}ms",
+        reportInitialDelay,
+        reportInterval);
+
+    service.scheduleAtFixedRate(
+        () -> {
+          try {
+            long usedDirectMemory = PlatformDependent.usedDirectMemory();
```

Comment:
What's the difference with the metric of `jvm_memory_bytes_used nonheap`

Reply:
We found "jvm_memory_bytes_used nonheap" is not correct, and it is small than direct memory. 
When OutOfDirectMemory is throw，"jvm_memory_bytes_used nonheap" is small and not correct.

Reply:
![image](https://github.com/apache/incubator-uniffle/assets/9440617/0cba772d-ae44-44f3-b7f2-379a6c072917)

![image](https://github.com/apache/incubator-uniffle/assets/9440617/02982217-75c4-483c-a4e1-fba97e5ca495)

Reply:
Got it.